stan/math/rev/fun/norm2.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef STAN_MATH_REV_FUN_NORM2_HPP
+#define STAN_MATH_REV_FUN_NORM2_HPP
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/core/typedefs.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/sign.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
```

Comment:
You can use the same recommendations that I made for the `norm1` headers here as well, should simplify the code a little

Reply:
Done.